core/src/main/java/io/questdb/cairo/wal/WalUtils.java

```
@@ -96,4 +106,73 @@ public static void createTxnLogFile(FilesFacade ff, MemoryMARW mem, Path txnSeqD
+                        final int walId = mem.getInt(offset + TableTransactionLogFile.TX_LOG_WAL_ID_OFFSET);
+                        final int segmentId = mem.getInt(offset + TableTransactionLogFile.TX_LOG_SEGMENT_OFFSET);
+                        final int segmentTxn = mem.getInt(offset + TableTransactionLogFile.TX_LOG_SEGMENT_TXN_OFFSET);
+                        if (walId > 0) {
+                            tablePath.concat(WAL_NAME_BASE).put(walId).slash().put(segmentId);
+                            try (WalEventReader eventReader = walEventReader) {
+                                WalEventCursor walEventCursor = eventReader.of(tablePath, segmentTxn);
+                                if (walEventCursor.getType() == MAT_VIEW_DATA) {
+                                    return walEventCursor.getDataInfoExt().getLastRefreshBaseTableTxn();
+                                }
+                                if (walEventCursor.getType() == MAT_VIEW_INVALIDATE) {
+                                    return txnInvalid;
+                                }
+                            } catch (Throwable th) {
+                                // walEventReader may not be able to find/open the WAL-e files
+                                try (BlockFileReader blockReader = blockFileReader) {
+                                    tablePath.trimTo(tablePathLen).concat(MatViewState.MAT_VIEW_STATE_FILE_NAME);
+                                    blockReader.of(tablePath.$());
+                                    matViewStateReader.of(blockReader, tableToken);
+                                    // read from state file if exists
+                                    if (!matViewStateReader.isInvalid()) {
+                                        return matViewStateReader.getLastRefreshBaseTxn();
+                                    }
+                                } catch (Throwable th2) {
```

We shouldn't swallow this exception, but instead re-throw it and log an error message in the engine. Otherwise, there will be no logs in case, for example, when the state file was deleted externally.